### ISSUE TITLE
Support for acr dedicated agent pool in azurerm_container_registry

### ISSUE THREAD

**Issue by benjamin37** [hashicorp/terraform-provider-azurerm#15936]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* azurerm_container_registry

### Potential Terraform Configuration

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_virtual_network" "example" {
  name                = "example-vnet"
  address_space       = ["10.0.0.0/16"]
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_subnet" "example" {
  name                 = "example-subnet"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example.name
  address_prefixes     = ["10.0.1.0/24"]
  }
}

resource "azurerm_container_registry" "example" {
  name                = "containerRegistry1"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  
  dedicated_agent_pool {
    name = "agentPool"
    tier = "S2"
    subnet_id = azurerm_subnet.example.id
    count = 2
  }
}
```

### References

* [acr dedicated agent pool](https://docs.microsoft.com/en-us/azure/container-registry/tasks-agent-pools)
* [azure cli commands](https://docs.microsoft.com/de-de/cli/azure/acr/agentpool?view=azure-cli-latest)

**Comment by tombuildsstuff** [comment]
x-ref: https://github.com/hashicorp/terraform-provider-azurerm/pull/15044
x-ref: https://github.com/hashicorp/terraform-provider-azurerm/pull/14790

**Comment by mgugino-pdig** [comment]
FYI, I believe this has been implemented here and is now available in several releases: https://github.com/hashicorp/terraform-provider-azurerm/pull/16258

This issue can likely be closed.

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #16258. As such, I am going to mark this issue as closed. If that is not the case, please provide additional information including the version in which you are still experiencing this issue, thanks!
